### ISSUE TITLE
How to populate dropdown in ejs

### ISSUE THREAD

**Issue by paragshah1979** [mde/ejs#422]
Hello,
   I have a requirement where i want to populate dropdown field where field values should come from mongodb collection.
  I have done below code in my js file which finds the data from mongodb collection and renders it to client.

var mongoose = require("mongoose");
var Employee = require("../models/Employee");
var Areas = require("../models/Areas");
var employeeController = {};

employeeController.create = function(req, res) {
	// Fetch area and send to create page
			Areas.find({}).exec(function (err, Areas) {

					if (err) {
						console.log("Error:", err);
					}
					console.log("Areas  :"+ Areas);
			res.render("../views/employees/create", {Area: Area});		
			});
};

Above code successfully gives query result in Area. 
Now my code in ejs file is as below:
<select id=”area” name=”area”> </select>
<% console.log('Area values are : ', Areas) %>
<script>
  var vArea = '<%= Areas %>';
      var select = document.getElementById(“area”), arr = [1,2,3,4];
      for (var I = 0; i< vArea.length; i++)
      {
        Var option = document.createElement(“OPTION”), txt= document.createTextNode(vArea[i]);
    Option.appendChile(txt);
     Option.setAttribute(“value”,vArea[i]);
    Select.insertBefore(option,select.lastChild);
 
      }
</script>  

Now, <% console.log('Area values are : ', Areas) %>  gives me correct values in console but when I assign the value vArea = '<%= Areas %>' then it returns me a null value. 

 Basically, I want to populate my <select id=”area” name=”area”> </select> and with values Areas which I rendered from my JS file.

Can you please share sample test code to achieve this.

Appreciate your response.

Thanks,
Parag

**Comment by paragshah1979** [comment]
[EmployeeController.txt](https://github.com/mde/ejs/files/2847082/EmployeeController.txt)

[create.txt](https://github.com/mde/ejs/files/2847086/create.txt)

Attached both the files for your reference.

**Comment by mde** [comment]
I'm not entirely sure what you're asking. This doesn't appear to be any sort of issue related to EJS itself. These issues for the EJS project are really just for things related to the library, not to help you debug your application code. If this is an actual problem with the EJS library, it helps to have a minimal failing test that demonstrates the problem you're having. (See the "tests" directory in the project to see how to write tests for EJS.)

**Comment by nfaltir** [comment]
xc
